### ISSUE TITLE
Custom snippets in snippets.cson ".text.html.php" work outside of <php ?> tags but not inside

### ISSUE THREAD

**Issue by QuentinMcCarthy** [atom/language-php#339]
### Prerequisites

* [X] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: http://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode
    * Followed all applicable steps in the debugging guide: http://flight-manual.atom.io/hacking-atom/sections/debugging/
    * Checked the FAQs on the message board for common solutions: https://discuss.atom.io/c/faq
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom
    * Checked that there is not already an Atom package that provides the described functionality: https://atom.io/packages

### Description

As stated in the title, creating a custom snippet for PHP. The snippets work outside of the php tags, but not inside.

### Steps to Reproduce

1. Create a snippet in the snippets.cson file under ".text.html.php"
2. Use snippet outside of php tags
3. Attempt to use snippet inside of php tags

**Expected behavior:** Snippet works correctly, overriding preset snippets as well.

**Actual behavior:** Preset snippets execute, custom snippets only execute outside of php tags.

**Reproduces how often:** 100%

### Versions

You can get this information from copy and pasting the output of `atom --version` and `apm --version` from the command line. Also, please include the OS and what version of the OS you're running.

Atom Version 1.30.0
OS Mac 10.13.4

### Additional Information

Any additional information, configuration or data that might be necessary to reproduce the issue.

**Comment by Ingramz** [comment]
I am unable to reproduce this - I tried overriding `echoh` and it replaces correctly in both conditions.

Would it be possible for you to provide the snippet you have in `snippets.cson`?

**Comment by QuentinMcCarthy** [comment]
Yup! I was trying to override the "foreach" loop snippet. I can provide screenshots of what happens, as well.

`
'.text.html.php':
  'foreach':
    'prefix': 'foreach'
    'body': """
		foreach (${1:$loop} as ${2:$loopItem}){
			$3
		}
	"""`

And this is execution of both in and out of the php tags

![image](https://user-images.githubusercontent.com/36647653/46629029-f83e4500-cb9b-11e8-95db-7d79c933dd84.png)

**Comment by Ingramz** [comment]
I see now. You'll have to define the snippet twice as the more specific scope always wins (if you take a look at the [source](https://github.com/atom/language-php/blob/master/snippets/language-php.cson), you can find `foreach` snipped defined there twice).

```cson
'.source.php':
  'foreach':
    'prefix': 'foreach'
    'body': """
      foreach (${1:$loop} as ${2:$loopItem}){
        $3
      }
    """
```

**Comment by QuentinMcCarthy** [comment]
Well that seems a little silly. Do I have to define it in .source.php? Or only in .text.html.php?

**Comment by Ingramz** [comment]
You need to define it in both, it is a bit silly, but this allows more granular overriding of such snippets, eg you want to only modify the version that works outside PHP tags.

There is a shortcut though, you can write `.text.html.php, .source.php` and it should work in both cases for that one snippet.
